lib/blue_state_digital/signup_form.rb

```diff
@@ -0,0 +1,100 @@
+module BlueStateDigital
+  class SignupFormField < ApiDataModel
+    FIELDS = [:id, :format, :label, :description, :is_required, :is_custom_field, :cons_field_id, :create_dt]
+    attr_accessor *FIELDS
+
+    # Takes a <signup_form_field> block already processed by Nokogiri
+    def self.from_xml(xml_record)
+      SignupFormField.new(id: xml_record[:id],
+                          format: xml_record.xpath('format').text,
+                          label: xml_record.xpath('label').text,
+                          description: xml_record.xpath('description').text,
+                          is_required: xml_record.xpath('is_required').text == '1',
+                          is_custom_field: xml_record.xpath('is_custom_field').text == '1',
+                          cons_field_id: xml_record.xpath('cons_field_id').text.to_i,
+                          create_dt: xml_record.xpath('create_dt').text)
+    end
+  end
+
+  class SignupForm < ApiDataModel
+    FIELDS = [:id, :name, :slug, :public_title, :create_dt]
+    attr_accessor *FIELDS
+
+    def self.clone(options)
+      clone_from_id = options[:clone_from_id]
+      slug = options[:slug]
+      name = options[:name]
+      public_title = options[:public_title]
+      connection = options[:connection]
+
+      params = {signup_form_id: clone_from_id, title: public_title, signup_form_name: name, slug: slug}
+      xml_response = connection.perform_request '/signup/clone_form', {}, 'POST', params
+
+      doc = Nokogiri::XML(xml_response)
+      record = doc.xpath('//signup_form').first
+      if record
+        id = record.xpath('id').text.to_i
+        SignupForm.new(id: id, name: name, slug: slug, public_title: public_title, connection: connection)
+      else
+        raise "Cloning signup form failed with message: #{xml_response}"
+      end
+    end
+
+    # Takes a hash like {field_data: {'field label' => 'field value'}, email_opt_in: false, source: 'foo', subsource: 'bar'}
+    def process_signup(data)
```

Comment:
`data` is a hash with a few specific keys (email_opt_in, source, subsource) and a hash of "field data" that corresponds to the fields on the form.  I tried to explain this in the comment above, but it didn't come out very clearly.  We could add a class but I'm not sure that would actually make it any clearer?